### PR TITLE
feat: add project aliases config

### LINKED ISSUE #12: Force override dependencies
There are some mods that depend on others, and when you do `pakku add <slug>` it will add the depencies too, thing is, sometimes the mod you are adding has an dependency on an old or deprecated mod that got updated, my example `dramatic-trees` is the new `dynamic-trees` for `1.12.2` (fork) and when I add the addons for it, lets say Quark Dynamic Trees, it adds `dynamic-trees` as a dependency and installs it.

Currently, I need to manually change the dependency ID and remove the wrongly added project, but running `pakku upade --all` will revert the changes.

Please make a way to prevent the manual changes to roll back, or better yet, provide a `pakku add <slug> --override_dependency <old dependency slug>:<new dependency slug>`.

Thanks for your work! Pakku is a great tool :D

### LINKED ISSUE #33: Project aliases
A config like https://sinytra.org/docs/connector/configuration#global-mod-aliases and https://fabricmc.net/wiki/documentation:fabric_mod_json#mod_loading `provides` is useful.

`--no-deps` isn't enough in this situation. A project is able to provide another project in Minecraft world

### PR DESCRIPTION
**Pull request by SettingDust**
fix #33
fix #12

### CODE REVIEW

**Review by juraj-hrivnak — CHANGES_REQUESTED** [review]
A project should inherit the `alias` &/or `aliases` from the ProjectConfig.

See [here](https://github.com/juraj-hrivnak/Pakku/blob/main/src/commonMain/kotlin/teksturepako/pakku/api/projects/Project.kt#L184).

File: src/commonMain/kotlin/teksturepako/pakku/api/data/LockFile.kt [hunk 1]
```diff
@@ -205,8 +205,9 @@ data class LockFile(
         return removed
     }
 
-    fun getProject(input: String): Project? = this.projects
-        .find { project -> input in project || project.files.any { input in it.fileName } }
+    fun getProject(input: String): Project? = (ConfigFile.readOrNull()?.projectAliases?.get(input) ?: input).let { transformedInput ->
+        this.projects.find { project -> transformedInput in project || project.files.any { transformedInput in it.fileName } }
+    }
```
[the remark this answers is not in the document]
Comment (juraj-hrivnak) [reply src/commonMain/kotlin/teksturepako/pakku/api/data/LockFile.kt:210]: Unnecessary with config inheritance.

File: src/commonMain/kotlin/teksturepako/pakku/cli/cmd/Add.kt [hunk 2]
```diff
@@ -110,7 +117,14 @@ class Add : CliktCommand()
...
                         {
-                            project.resolveDependencies(terminal, reqHandlers, lockFile, projectProvider, platforms)
+                            project.resolveDependencies(
+                                terminal,
+                                reqHandlers,
+                                lockFile,
+                                configFile,
+                                projectProvider,
+                                platforms
+                            )
```
[the remark this answers is not in the document]
Comment (juraj-hrivnak) [reply src/commonMain/kotlin/teksturepako/pakku/cli/cmd/Add.kt:127]: Also unnecessary with config inheritance.

[the remark this answers is not in the document]
Comment (SettingDust) [reply src/commonMain/kotlin/teksturepako/pakku/cli/cmd/Add.kt:127]: My IDEA is formatting it. Can you provide your `.editorconfig` by export it from IDEA?

File: src/commonMain/kotlin/teksturepako/pakku/cli/cmd/Cfg.kt [hunk 3]
```diff
@@ -56,6 +57,12 @@ class Cfg : CliktCommand()
     private val shadersPathOpt by option("--shaders-path", metavar = "path")
         .help("Change the path for the `${ProjectType.SHADER}` project type")
 
+    private val projectAliasesOpts: Map<String, String>? by option(
+        "-pa", "--project-alias",
```
[the remark this answers is not in the document]
Comment (SettingDust) [reply src/commonMain/kotlin/teksturepako/pakku/cli/cmd/Cfg.kt:61]: It's working XD. But I will remove this

File: src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt [hunk 4]
```diff
@@ -34,16 +34,19 @@ data class ConfigFile(
     private val overrides: MutableList<String> = mutableListOf(),
 
     /** A mutable list of server overrides packed up with the modpack. */
-    @SerialName("server_overrides") private val serverOverrides: MutableList<String> = mutableListOf(),
+    private val serverOverrides: MutableList<String> = mutableListOf(),
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt:37]: Serial name is missing

File: src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt [hunk 5]
```diff
@@ -34,16 +34,19 @@ data class ConfigFile(
...
 
     /** A mutable list of client overrides packed up with the modpack. */
-    @SerialName("client_overrides") private val clientOverrides: MutableList<String> = mutableListOf(),
+    private val clientOverrides: MutableList<String> = mutableListOf(),
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt:40]: Same here

File: src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt [hunk 6]
```diff
@@ -34,16 +34,19 @@ data class ConfigFile(
...
 
     /**  A map of project types to their respective paths. */
     val paths: MutableMap<String, String> = mutableMapOf(),
 
     /** A mutable map of _project slugs, names, IDs or filenames_ to _project configs_. */
-    val projects: MutableMap<String, ProjectConfig> = mutableMapOf()
+    val projects: MutableMap<String, ProjectConfig> = mutableMapOf(),
+
+    /** A mutable map of project aliases to their respective project slugs. */
+    val projectAliases: MutableMap<String, String> = mutableMapOf()
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt:49]: This shouldn't be needed anymore

File: src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt [hunk 7]
```diff
@@ -110,9 +113,10 @@ data class ConfigFile(
     data class ProjectConfig(
         var type: ProjectType? = null,
         var side: ProjectSide? = null,
-        @SerialName("update_strategy") var updateStrategy: UpdateStrategy? = null,
-        @SerialName("redistributable") var redistributable: Boolean? = null,
-        var subpath: String? = null
+        var updateStrategy: UpdateStrategy? = null,
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt:116]: Missing serial name

File: src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt [hunk 8]
```diff
@@ -110,9 +113,10 @@ data class ConfigFile(
...
+        var redistributable: Boolean? = null,
+        var subpath: String? = null,
+        var aliases :MutableSet<String>? = null
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/data/ConfigFile.kt:119]: Formating: `var aliases: MutableSet<String>? = null`

File: src/commonMain/kotlin/teksturepako/pakku/api/data/Json.kt [hunk 9]
```diff
@@ -2,6 +2,7 @@ package teksturepako.pakku.api.data
 
 import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonNamingStrategy
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/data/Json.kt:5]: Unused import

File: src/commonMain/kotlin/teksturepako/pakku/api/projects/Project.kt [hunk 10]
```diff
@@ -43,6 +43,7 @@ data class Project(
     @SerialName("redistributable") var redistributable: Boolean = true,
 
     private var subpath: String? = null,
+    var aliases: MutableSet<String> = mutableSetOf(),
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/projects/Project.kt:46]: Should be nullable and null by default.

**Review by juraj-hrivnak — CHANGES_REQUESTED** [review]

**Review by juraj-hrivnak — APPROVED** [review]
Great job! Thanks!

**Pull request merged**